e2e-test/fluxd.yaml

```diff
@@ -0,0 +1,166 @@
+kind: Secret
+metadata:
+  creationTimestamp: null
+  name: flux-git-deploy
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  labels:
+    name: flux
+  name: flux
+  namespace: default
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  labels:
+    name: flux
+  name: flux
+rules:
+  - apiGroups: ["*"]
+    resources: ["*"]
+    verbs: ["*"]
+  - nonResourceURLs: ["*"]
+    verbs: ["*"]
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRoleBinding
+metadata:
+  labels:
+    name: flux
+  name: flux
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: flux
+subjects:
+  - kind: ServiceAccount
+    name: flux
+    namespace: default
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: flux-recv
+data:
+  fluxrecv.yaml: |
+    fluxRecvVersion: 1
+    endpoints: []
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: flux
+  namespace: default
+  labels:
+    name: flux
+    type: deployment
+spec:
+  replicas: 1
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 0
+      maxSurge: 1
+  selector:
+    matchLabels:
+      name: flux
+  template:
+    metadata:
+      labels:
+        name: flux
+        type: deployment
+    spec:
+      serviceAccountName: flux
+      volumes:
+        - name: git-key
+          secret:
+            secretName: flux-git-deploy
+            defaultMode: 0400 # when mounted read-only, we won't be able to chmodª
+        - name: fluxrecv-config
+          configMap:
+            name: flux-recv
+            defaultMode: 0400
+        - name: git-keygen
+          emptyDir:
+            medium: Memory
+        - name: tmp
+          emptyDir:
+            medium: Memory
+        - name: kubedir
+          configMap:
+            name: flux-kube-config
+        - name: sourcegitrepo
+          hostPath:
+            path: /sourcegitrepo
+      containers:
+        - name: flux
+          image: docker.io/fluxcd/flux:1.18.0
+          imagePullPolicy: IfNotPresent
+          resources:
+            requests:
+              cpu: 50m
+              memory: 64Mi
+          ports:
+            - containerPort: 3030 # informational
+          env:
+            - name: KUBECONFIG
+              value: /root/.kubectl/config
+          volumeMounts:
+            # - name: git-key
+            #   mountPath: /etc/fluxd/ssh
+            #   readOnly: true
+            - name: git-keygen
+              mountPath: /var/fluxd/keygen
+            - name: tmp
+              mountPath: /tmp
```

Comment:
This is not used if I read the `args` right?

Reply:
Not sure. I stole it from our lunar-way-flux setup

Reply:
Ah yes. Flux does its operations on temporary directories, so this is still relevant 👍